[LBSE] Begin stub implementation of RenderSVGViewportContainer

nikolaszimmermann opened the pull request:
#### 273d610500568994a594191403c869ffe26098e2
<pre>
[LBSE] Begin stub implementation of RenderSVGViewportContainer
<a href="https://bugs.webkit.org/show_bug.cgi?id=242701">https://bugs.webkit.org/show_bug.cgi?id=242701</a>

Reviewed by Rob Buis.

Rename LegacyRenderSVGViewportContainer -&gt; RenderSVGViewportContainer and adapt all call sites.
Introduce layer-aware RenderSVGViewportContainer at the same time as stub implementation.

Follow-up patches will bring support for inner &lt;svg&gt; elements.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isLegacySVGViewportContainer const):
* Source/WebCore/rendering/svg/LegacyRenderSVGContainer.cpp:
(WebCore::LegacyRenderSVGContainer::layout):
(WebCore::LegacyRenderSVGContainer::paint):
(WebCore::LegacyRenderSVGContainer::nodeAtFloatPoint):
* Source/WebCore/rendering/svg/LegacyRenderSVGViewportContainer.cpp: Copied from Source/WebCore/rendering/svg/RenderSVGViewportContainer.cpp.
(WebCore::LegacyRenderSVGViewportContainer::LegacyRenderSVGViewportContainer):
(WebCore::LegacyRenderSVGViewportContainer::svgSVGElement const):
(WebCore::LegacyRenderSVGViewportContainer::determineIfLayoutSizeChanged):
(WebCore::LegacyRenderSVGViewportContainer::applyViewportClip):
(WebCore::LegacyRenderSVGViewportContainer::calcViewport):
(WebCore::LegacyRenderSVGViewportContainer::calculateLocalTransform):
(WebCore::LegacyRenderSVGViewportContainer::viewportTransform const):
(WebCore::LegacyRenderSVGViewportContainer::pointIsInsideViewportClip):
(WebCore::LegacyRenderSVGViewportContainer::paint):
* Source/WebCore/rendering/svg/LegacyRenderSVGViewportContainer.h: Copied from Source/WebCore/rendering/svg/RenderSVGViewportContainer.h.
* Source/WebCore/rendering/svg/RenderSVGViewportContainer.cpp:
(WebCore::RenderSVGViewportContainer::RenderSVGViewportContainer):
(WebCore::RenderSVGViewportContainer::svgSVGElement const):
(WebCore::RenderSVGViewportContainer::computeViewport const):
(WebCore::RenderSVGViewportContainer::updateLayerTransform):
(WebCore::RenderSVGViewportContainer::applyTransform const):
(WebCore::RenderSVGViewportContainer::determineIfLayoutSizeChanged): Deleted.
(WebCore::RenderSVGViewportContainer::applyViewportClip): Deleted.
(WebCore::RenderSVGViewportContainer::calcViewport): Deleted.
(WebCore::RenderSVGViewportContainer::calculateLocalTransform): Deleted.
(WebCore::RenderSVGViewportContainer::viewportTransform const): Deleted.
(WebCore::RenderSVGViewportContainer::pointIsInsideViewportClip): Deleted.
(WebCore::RenderSVGViewportContainer::paint): Deleted.
* Source/WebCore/rendering/svg/RenderSVGViewportContainer.h:
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
(WebCore::layoutSizeOfNearestViewportChanged):
(WebCore::SVGRenderSupport::transformToRootChanged):
* Source/WebCore/svg/SVGLengthContext.cpp:
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::createElementRenderer):
(WebCore::SVGSVGElement::currentViewportSize const):

Canonical link: <a href="https://commits.webkit.org/252500@main">https://commits.webkit.org/252500@main</a>
</pre>

rwlbuis commented on Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
```diff
@@ -207,17 +207,13 @@ static inline void invalidateResourcesOfChildren(RenderElement& renderer)
 {
     const RenderElement* start = &renderer;
-    while (start && !start->isSVGRootOrLegacySVGRoot() && !is<RenderSVGViewportContainer>(*start))
+    while (start && !is<LegacyRenderSVGRoot>(*start) && !is<LegacyRenderSVGViewportContainer>(*start))
```
Why the first part of this change?

nikolaszimmermann replied:
Ah, SVGRenderSupport is not used at all for LBSE -- thus handling RenderSVGRoot is not necessary.

rwlbuis reviewed (changes requested):
Added a question.

rwlbuis reviewed (approved):
LGTM.

nikolaszimmermann committed:
[LBSE] Begin stub implementation of RenderSVGViewportContainer
https://bugs.webkit.org/show_bug.cgi?id=242701

Reviewed by Rob Buis.

Rename LegacyRenderSVGViewportContainer -> RenderSVGViewportContainer and adapt all call sites.
Introduce layer-aware RenderSVGViewportContainer at the same time as stub implementation.

Follow-up patches will bring support for inner <svg> elements.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isLegacySVGViewportContainer const):
* Source/WebCore/rendering/svg/LegacyRenderSVGContainer.cpp:
(WebCore::LegacyRenderSVGContainer::layout):
(WebCore::LegacyRenderSVGContainer::paint):
(WebCore::LegacyRenderSVGContainer::nodeAtFloatPoint):
* Source/WebCore/rendering/svg/LegacyRenderSVGViewportContainer.cpp: Copied from Source/WebCore/rendering/svg/RenderSVGViewportContainer.cpp.
(WebCore::LegacyRenderSVGViewportContainer::LegacyRenderSVGViewportContainer):
(WebCore::LegacyRenderSVGViewportContainer::svgSVGElement const):
(WebCore::LegacyRenderSVGViewportContainer::determineIfLayoutSizeChanged):
(WebCore::LegacyRenderSVGViewportContainer::applyViewportClip):
(WebCore::LegacyRenderSVGViewportContainer::calcViewport):
(WebCore::LegacyRenderSVGViewportContainer::calculateLocalTransform):
(WebCore::LegacyRenderSVGViewportContainer::viewportTransform const):
(WebCore::LegacyRenderSVGViewportContainer::pointIsInsideViewportClip):
(WebCore::LegacyRenderSVGViewportContainer::paint):
* Source/WebCore/rendering/svg/LegacyRenderSVGViewportContainer.h: Copied from Source/WebCore/rendering/svg/RenderSVGViewportContainer.h.
* Source/WebCore/rendering/svg/RenderSVGViewportContainer.cpp:
(WebCore::RenderSVGViewportContainer::RenderSVGViewportContainer):
(WebCore::RenderSVGViewportContainer::svgSVGElement const):
(WebCore::RenderSVGViewportContainer::computeViewport const):
(WebCore::RenderSVGViewportContainer::updateLayerTransform):
(WebCore::RenderSVGViewportContainer::applyTransform const):
(WebCore::RenderSVGViewportContainer::determineIfLayoutSizeChanged): Deleted.
(WebCore::RenderSVGViewportContainer::applyViewportClip): Deleted.
(WebCore::RenderSVGViewportContainer::calcViewport): Deleted.
(WebCore::RenderSVGViewportContainer::calculateLocalTransform): Deleted.
(WebCore::RenderSVGViewportContainer::viewportTransform const): Deleted.
(WebCore::RenderSVGViewportContainer::pointIsInsideViewportClip): Deleted.
(WebCore::RenderSVGViewportContainer::paint): Deleted.
* Source/WebCore/rendering/svg/RenderSVGViewportContainer.h:
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
(WebCore::layoutSizeOfNearestViewportChanged):
(WebCore::SVGRenderSupport::transformToRootChanged):
* Source/WebCore/svg/SVGLengthContext.cpp:
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::createElementRenderer):
(WebCore::SVGSVGElement::currentViewportSize const):

Canonical link: https://commits.webkit.org/252500@main
